packages/core/src/store/level-db-store.ts

```diff
@@ -45,7 +45,18 @@ class LevelDBStoreMap {
     if (fs) {
       fs.mkdirSync(storePath, { recursive: true }) // create dir if it doesn't exist
     }
-    this.#map.set(fullLocation, new LevelC(storePath))
+
+    const levelDb = new LevelC(storePath)
+    // level-ts does not have an error handling exposed for opening errors. I access the private DB variable to add callbacks for logging.
+    // @ts-ignore private field
+    levelDb.DB.on('error', (err) => {
+      this.logger.warn(
+        `Received error when starting up leveldb at ${storePath} using level-ts: ${err}`
```

Comment:
Will error only occur on startup, or could this occur at any time, and should be indicated as a runtime error?

Reply:
I want to say only on startup when i look at the libraries, but I'm not 100% on that. From my crude tests, I haven't seen this get triggered outside of startup errors. Level-ts has "Promise forcing" so there should be no callback sused and therefore error events should not be used.